src/browser/browser.ts

```diff
@@ -4,56 +4,120 @@ import { Page } from "./page";
 import { BrowserMode } from "../types/browser/browser.types";
 import { browserContext } from "./browserUtils";
 import { Agent } from "../agent";
+import { Logger } from "../utils";
 
+/**
+ * Represents a browser session using Puppeteer.
+ * Manages the creation of new browser pages and handles the browser instance.
+ *
+ * @class
+ * @property {PuppeteerBrowser} browser - The Puppeteer browser instance.
+ * @property {BrowserMode} mode - The mode of the browser which could be headless or non-headless.
+ * @property {Agent} agent - The agent instance that interacts with the browser.
+ * @property {Logger} [logger] - Optional logger for logging browser activities.
+ * @property {string} userDataDir - The directory path for storing user data. Defaults to "/tmp".
+ */
 export class Browser {
-  // @ts-ignore
-  private browser: PuppeteerBrowser;
-  // @ts-ignore
+  private apiKey: string | undefined;
+  private endpoint: string | undefined;
+  private _pages: Page[] = [];
+
+  browser: PuppeteerBrowser;
   mode: BrowserMode;
-  // @ts-ignore
   agent: Agent;
+  logger?: Logger;
   private userDataDir = "/tmp"; // TODO: make this configurable
 
-  constructor() {}
-
-  private async init(
+  /**
+   * Initializes a new instance of the Browser class.
+   *
+   * @param {PuppeteerBrowser} browser - The Puppeteer browser instance.
+   * @param {Agent} agent - The agent instance that interacts with the browser.
+   * @param {BrowserMode} mode - The mode of the browser (e.g., headless).
+   * @param {Logger} [logger] - Optional logger for logging browser activities.
+   */
+  constructor(
     browser: PuppeteerBrowser,
     agent: Agent,
-    mode: BrowserMode
+    logger?: Logger,
+    opts?: { mode?: BrowserMode; apiKey?: string; endpoint?: string }
   ) {
     this.agent = agent;
     this.browser = browser;
-    this.mode = mode;
+    this.mode = opts?.mode || BrowserMode.text;
+    this.logger = logger;
+
+    this.apiKey = opts?.apiKey || process.env.HDR_API_KEY;
+    this.endpoint = opts?.endpoint || process.env.HDR_ENDPOINT;
   }
 
-  static async create(
+  /**
+   * Asynchronously launch a new Browser instance with given configuration.
+   *
+   * @param {boolean} headless - Specifies if the browser should be launched in headless mode.
+   * @param {Agent} agent - The agent that will interact with the browser.
+   * @param {Logger} [logger] - Optional logger to pass for browser operation logs.
+   * @param {string} [browserWSEndpoint] - The WebSocket endpoint to connect to a browser instance.
+   * @param {string[]} [browserLaunchArgs] - Additional arguments for launching the browser.
+   * @param {BrowserMode} [mode=BrowserMode.text] - The mode of the browser, defaults to text.
+   * @returns {Promise<Browser>} A promise that resolves to an instance of Browser.
+   */
+  static async launch(
```

Comment:
Why are we renaming this? I am not sure what this accomplishes other than forcing everyone to change the method they use, when we seem to be keeping AgentBrowser so as to not do that.

Reply:
trying to keep the puppeteer API as closely as possible https://github.com/puppeteer/puppeteer/blob/d8654f1eed657bb25da5d7167ed0abedf007d7a1/examples/pdf.js#L12

Reply:
Okay. Well let me at least patch the `bin` folder here so our alpha package doesn't crash when you try to use it, I suppose.